core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -79,6 +85,8 @@ public void commit(Snapshot snapshot) throws CommitException, UnknownTransaction
       throw e;
     }
 
+    snapshotHandleFuture.ifPresent(SnapshotHandleFuture::get);
```

Comment:
@feeblefakie 

> Can we just set Function or BiFunction like it does with `setSnapshotHandler` and invoke it here?

Yeah, good point. I think using `Function` should also work.

I named the interface `SnapshotHandle**Future**` since I wanted to encourage `setSnapshotHandler` users to implement asynchronous handlers that don't block `prepare()` and `validate()`. Also, the reason why I wrapped `Future` with `SnapshotHandleFuture` is for maintainability considering future modifications just in case. But, I'm okay to go with simple `Function`.

> Also, I feel SnapshotHandler sounds too general.

Ah, I agree. `BeforeCommitSnapshotHook` seems good. But it might sound like "validated" snapshots. How about `PreparedSnapshotHook`? Either is fine with me, thought.

Reply:
Thanks for the explanation.
Using `Future` for a non-blocking hook makes sense in this case since Snapshot won't be changed after the prepare phase, so I'm fine with it.
For the name, how about `AfterPrepareSnapshotHook`?
I'm also fine with `BeforeCommitSnapshotHook` and `PreparedSnapshotHook`.
I think you can decide.

Reply:
@feeblefakie Thanks for the comment!

> For the name, how about AfterPrepareSnapshotHook?
> I'm also fine with BeforeCommitSnapshotHook and PreparedSnapshotHook.

Strictly speaking, snapshots are sent asynchronously before `prepare()` completes. So, I hesitate a bit to use the terms `before` or `after` that emphasizes the order. `PreparedSnapshotHook` might imply that snapshots are **already prepared**, but I think it can be also interpreted that snapshots are in the process of being prepared, and relatively better. I'll go with `PreparedSnapshotHook`.

Reply:
Addressed it in https://github.com/scalar-labs/scalardb/pull/2271/commits/037913934c5e7dca44262c280b48136300fdeb92